app/src/main/java/de/triology/cas/services/RegistryLocal.java

```diff
@@ -217,31 +222,73 @@ public URI getCasLogoutUri(String doguname) throws GetCasLogoutUriException {
+        WatchService watchService;
+        try {
+            watchService = initializeWatchService(path);
+            var previousServiceAccounts = readServiceAccounts();
+            WatchKey key;
+            LOGGER.info("wait for changes under {}", LOCAL_CONFIG_DIR);
+            while ((key = watchService.take()) != null) { // take does not return null
+                if (!key.isValid()) {
+                    LOGGER.info("watch key was cancelled or watch service was closed");
+                    LOGGER.info("try to reregister watch");
+                    watchService = initializeWatchService(path);
+                    continue;
+                }
+
+                List<WatchEvent<?>> watchEvents = key.pollEvents();
+                boolean serviceAccountRegistryChanged = false;
+                for (WatchEvent<?> event : watchEvents) {
+                    if (event.context().equals(LOCAL_CONFIG_FILE_NAME)) {
+                        serviceAccountRegistryChanged = true;
+                        break;
                     }
                 }
-            } catch (IOException e) {
-                throw new RegistryException("Failed to addDoguChangeListener", e);
-            } catch (InterruptedException e) {
-                Thread.currentThread().interrupt();
-                throw new RegistryException("Failed to addDoguChangeListener", e);
+
+                if (!serviceAccountRegistryChanged) {
+                    continue;
+                }
+
+                var currentServiceAccounts = readServiceAccounts();
+                if (!previousServiceAccounts.deepEquals(currentServiceAccounts)) {
+                    LOGGER.info("services changed. call doguChangeListener");
+                    doguChangeListener.onChange();
+                    previousServiceAccounts = currentServiceAccounts;
+                }
+
+                // Resetting the watchKey is very important. Otherwise, the key returned from take() (or poll()) will not return any more events.
+                if (!key.reset()) {
+                    LOGGER.info("reset key is no longer valid");
+                    LOGGER.info("try to reregister watch");
+                    watchService = initializeWatchService(path);
+                }
             }
-        });
-        t1.start();
+        } catch (IOException e) {
+            throw new RegistryException("Failed to addDoguChangeListener", e);
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new RegistryException("Failed to addDoguChangeListener", e);
+        }
+    }
+
+    private WatchService initializeWatchService(Path path) throws IOException {
+        WatchService watchService = fileSystem.newWatchService();
+        registerServiceRegistryWatch(path, watchService);
+
+        return watchService;
+    }
+
+    private void registerServiceRegistryWatch(Path path, WatchService watchService) throws IOException {
```

Comment:
I don't really see the point of this method (eventhough I call myself a fan of small methods). It's just a void-one liner that is called once. It's subjective, so you can decide, but I would inline this method.